azurefunctions/src/main/java/com/microsoft/durabletask/azurefunctions/DurableClientContext.java

```diff
@@ -51,9 +56,55 @@ public DurableTaskClient getClient() {
             throw new IllegalStateException("The client context RPC base URL was invalid!", ex);
         }
 
-        return new DurableTaskGrpcClientBuilder().port(rpcURL.getPort()).build();
+        this.client = new DurableTaskGrpcClientBuilder().port(rpcURL.getPort()).build();
+        return this.client;
     }
 
+    /**
+     * Creates an HTTP response which either contains a payload of management URLs for a non-completed instance
+     * or contains the payload containing the output of the completed orchestration.
+     * <p>
+     * If the orchestration instance completes within the specified timeout, then the HTTP response payload will
+     * contains the output of the orchestration instance formatted as JSON. However, if the orchestration does not
+     * complete within the specified timeout, then the HTTP response will be identical to that of the
+     * {@link #createCheckStatusResponse(HttpRequestMessage, String)} API.
+     * </p>
+     * @param request the HTTP request that triggered the current function
+     * @param instanceId the unique ID of the instance to check
+     * @param timeout total allowed timeout for output from the durable function
+     * @return an HTTP response which may include a 202 and location header or a 200 with the durable function output in the response body
+     */
+    public HttpResponseMessage waitForCompletionOrCreateCheckStatusResponse(
+            HttpRequestMessage<?> request,
+            String instanceId,
+            Duration timeout) {
+        if (this.client == null) {
+            this.client = getClient();
+        }
+        OrchestrationMetadata orchestration;
+        try {
+            orchestration = this.client.waitForInstanceCompletion(instanceId, timeout, true);
+            return request.createResponseBuilder(HttpStatus.ACCEPTED)
+                    .header("Content-Type", "application/json")
+                    .body(orchestration.getSerializedOutput())
+                    .build();
+        } catch (TimeoutException e) {
+            return createCheckStatusResponse(request, instanceId);
+        }
+    }
+
+    /**
+     * Creates an HTTP response that is useful for checking the status of the specified instance.
+     * <p>
+     * The payload of the returned
+     * @see <a href="https://learn.microsoft.com/en-us/java/api/com.microsoft.azure.functions.httpresponsemessage?view=azure-java-stable">HttpResponseMessage</a>
```

Comment:
nit: it's a good practice to remove the language component of these URLs so that readers can see the docs in their native language if another local translation is available. As a bonus, it also makes the line a little shorter. :) We can remove the query string too.
```suggestion
     * @see <a href="https://learn.microsoft.com/java/api/com.microsoft.azure.functions.httpresponsemessage">HttpResponseMessage</a>
```